src/lib/utils.ts

```diff
@@ -79,3 +79,31 @@ export const getUserRank = (user: PublicProfile): string => {
  * @returns whether or not the source is a gif
  */
 export const isSrcAGif = (src: string): boolean => /\.gif($|&)/.test(src);
+
+const MONTHS = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
+
+/**
+ * Given ISO-8601 strings for the start and end date of a location,
+ * this function returns a formatted date for us to display on each event.
+ * Example outputs:
+ *  - Apr 25
+ *  - Jun 11
+ */
+export const formatDate = (date: string): string => {
+  const startDate = new Date(date);
+  const month = MONTHS[startDate.getMonth()];
+  return `${month} ${startDate.getUTCDate()}`;
+};
+
+export const formatTime = (start: string, end: string): string => {
+  const startTime = new Date(start).toLocaleTimeString('en-US', {
+    hour: 'numeric',
+    minute: '2-digit',
+  });
+  const endTime = new Date(end).toLocaleTimeString('en-US', { hour: 'numeric', minute: '2-digit' });
+  return `${startTime} - ${endTime}`;
+};
+
+export const formatEventDate = (start: string, end: string): string => {
+  return `${formatDate(start)}, ${formatTime(start, end)}`;
+};
```

Comment:
If the other functions aren't being used anywhere, then you could just use [`formatRange`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/formatRange):

```suggestion
const rangeFormat = new Intl.DateTimeFormat('en-US', { month: 'short', day: 'numeric', hour: 'numeric', minute: '2-digit' });
export const formatEventDate = (start: string, end: string): string => {
  return rangeFormat.formatRange(new Date(start), new Date(end));
};
```

This is what its output looks like; it's smart and will only repeat the date or am/pm if it's different:

```js
> new Intl.DateTimeFormat('en-US', { month: 'short', day: 'numeric', hour: 'numeric', minute: '2-digit' }).formatRange(new Date(2021, 9, 21, 18, 0), new Date(2021, 9, 21, 18, 30))
'Oct 21, 6:00 – 6:30 PM'
```